conn.go

```diff
@@ -461,17 +660,76 @@ func (c *Conn) handleRcpt(arg string) {
 		c.writeResponse(501, EnhancedCode{5, 5, 2}, "Was expecting RCPT arg syntax of TO:<address>")
 		return
 	}
-	if len(strings.Fields(p.s)) > 0 {
-		c.writeResponse(501, EnhancedCode{5, 5, 2}, "RCPT parameters are not supported")
-		return
-	}
 
 	if c.server.MaxRecipients > 0 && len(c.recipients) >= c.server.MaxRecipients {
 		c.writeResponse(452, EnhancedCode{4, 5, 3}, fmt.Sprintf("Maximum limit of %v recipients reached", c.server.MaxRecipients))
 		return
 	}
 
+	args, err := parseArgs(p.s)
+	if err != nil {
+		c.writeResponse(501, EnhancedCode{5, 5, 4}, "Unable to parse RCPT ESMTP parameters")
+		return
+	}
+
 	opts := &RcptOptions{}
+
+	for key, value := range args {
+		switch key {
+		case "NOTIFY":
+			if !c.server.EnableDSN {
+				c.writeResponse(504, EnhancedCode{5, 5, 4}, "NOTIFY is not implemented")
+				return
+			}
+			notify := []DSNNotify{}
+			seen := map[DSNNotify]struct{}{}
+			for _, val := range strings.Split(value, ",") {
+				val = strings.ToUpper(val)
+				switch DSNNotify(val) {
+				case NotifyNever:
+					if len(seen) != 0 {
```

Comment:
Maybe we can move this check at the end of the loop, to avoid duplicating them?

```go
for _, val := range strings.Split(value, ",") {
    …
}
if _, ok := seen[NotifyNever]; ok && len(seen) > 1 {
    // error
}
```

Reply:
Done!